timeseries-streaming/timeseries-python-applications/MLPipeline/timeseries/encoder_decoder/encoder_decoder_preprocessing.py

```diff
@@ -12,70 +12,49 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-from typing import Dict, Text, Any
+import math
+from datetime import datetime
+from typing import Dict, Text, Any, Optional
 
 import tensorflow as tf
+import tensorflow_transform as tft
+from timeseries.utils import timeseries_transform_utils
 
 
-def preprocessing_fn(inputs: Dict[Text, Any]) -> Dict[Text, Any]:
+def preprocessing_fn(inputs: Dict[Text, Any], custom_config: Dict[Text, Any]) -> Dict[Text, Any]:
     """tf.transform's callback function for preprocessing inputs.
 
     Args:
       inputs: map from feature keys to raw not-yet-transformed features.
+      custom_config:
+        timesteps: The number of timesteps in the look back window
+        features: Which of the features from the TF.Example to use in the model.
 
     Returns:
       Map from string feature key to transformed feature operations.
     """
+    timesteps = custom_config['timesteps']
```

Comment:
this custom config is user provided? I saw the code is different [here](https://github.com/GoogleCloudPlatform/dataflow-sample-applications/blob/master/timeseries-streaming/timeseries-python-applications/MLPipeline/timeseries/encoder_decoder/encoder_decoder_preprocessing.py#L40)

which one is head?

Reply:
Yes with the upgrade to 0.23.0 we can now have the user pass this information into preprocessing_fn without the need to do the Key based workaround.